packages/@ngtools/webpack/src/loader.ts

```diff
@@ -435,6 +435,77 @@ function _diagnoseDeps(reasons: ModuleReason[], plugin: AotPlugin, checked: Set<
 }
 
 
+export function _getModuleExports(plugin: AotPlugin,
+                                  refactor: TypeScriptFileRefactor): ts.Identifier[] {
+  const exports = refactor
+    .findAstNodes(refactor.sourceFile, ts.SyntaxKind.ExportDeclaration, true);
+
+  return exports
+    .filter(node => {
+
+      const identifiers = refactor.findAstNodes(node, ts.SyntaxKind.Identifier, false);
+
+      identifiers
+        .filter(node => node.getText() === plugin.entryModule.className);
+
+      return identifiers.length > 0;
+    }) as ts.Identifier[];
+}
+
+
+export function _replaceExport(plugin: AotPlugin, refactor: TypeScriptFileRefactor) {
+  if (!plugin.replaceExport) {
+    return;
+  }
+  _getModuleExports(plugin, refactor)
+    .forEach(node => {
+      const factoryPath = _getNgFactoryPath(plugin, refactor);
+      const factoryClassName = plugin.entryModule.className + 'NgFactory';
+      const exportStatement = `export \{ ${factoryClassName} \} from '${factoryPath}'`;
+      refactor.appendAfter(node, exportStatement);
+    });
+}
+
+
+export function _exportModuleMap(plugin: AotPlugin, refactor: TypeScriptFileRefactor) {
+  if (!plugin.replaceExport) {
+    return;
+  }
+
+  _getModuleExports(plugin, refactor)
+    .forEach(node => {
+      const routes: any[] = Object.keys(plugin.discoveredLazyRoutes)
+        .map(routeKey => {
+          let lazyRouteKey = routeKey.split('#')[0];
+          if (!plugin.skipCodeGeneration) {
+            lazyRouteKey += '.ngfactory';
+          }
+          return {
+            loadChildrenString: routeKey,
+            path: plugin.lazyRoutes[lazyRouteKey],
+            name: routeKey.split('#')[1]
+          };
+        });
+      let map: { [key: string]: string } = {};
+      routes.forEach(module => {
+        const append = plugin.skipCodeGeneration ? '' : 'NgFactory';
+        map[module.loadChildrenString] = module.name + append;
+      });
+
+      const dirName = path.normalize(path.dirname(refactor.fileName));
+      let exportStatement = `export const moduleMap = ${JSON.stringify(map)};`;
```

Comment:
Change this to not use `JSON.stringify` and instead properly build up the export map without string replacement hacks.